config/get_ginkgo_omp_config.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include <ginkgo/core/base/version.hpp>
+#include <cstdio>
+#include <string.h>
+
+int main()
+{
+   gko::version_info gko_version = gko::version_info::get();
+   if (strcmp(gko_version.omp_version.tag, "not compiled") == 0)
```

Comment:
Maybe I am missing something here, and not fully following the discussion in https://github.com/mfem/mfem/issues/3909. But, why do we need a test program and the compile time `MFEM_GINKGO_WITH_OMP` configuration variable if we can just use `gko::version_info::get` along with the MFEM `Device` configuration to determine whether to use `gko::ReferenceExecutor::create` vs. `gko::OmpExecutor::create` at runtime?

For example, with no compile-time configuration, I am thinking of the following type of thing in the `GinkgoExecutor` constructor:

```
gko::version_info gko_version = gko::version_info::get();
bool ginko_use_openmp = (strcmp(gko_version.omp_version.tag, "not compiled") == 0)
if (Device::Allows(Backend::OMP_MASK) && ginko_use_omp)
{
   executor = gko::OmpExecutor::create();
}
else
{
   executor = gko::ReferenceExecutor::create();
}
```

Similar logic could be applied for CUDA/HIP device support. Basically it seems like everything can be done at runtime here.

Reply:
Mostly I just figured, if we could settle the issue at compile time, why not go ahead and do it? But I guess it doesn't really matter, as it's not critical for performance or anything. I can switch to a runtime check in the constructors if that would be preferred. It would certainly avoid adding complexity to the build system.

Reply:
I'm in favor of keeping the build system as simple as possible (only a matter of time before users come asking about errors encountered during `try_run`). This seems like a good fit for runtime configuration and is consistent with the rest of the MFEM approach of device configuration at run time, not build time.